dorado/routines.py

```diff
@@ -215,11 +215,15 @@ def unsteady_plots(params, num_steps, timestep,
             raise ValueError('Output datatype/structure unsupported, modify'
                              ' the output reading portion of the code')
 
-        # then define the particle class and continue
-        particle = Particle(params)
+        # then define the particles class and continue
+        particle = Particles(params)
+        # generate some particles
+        if i == 0:
+            particle.generate_particles(Np_tracer, seed_xloc, seed_yloc)
+        else:
+            particle.generate_particles(0, [], [], 'random', walk_data)
 
-        walk_data = particle.run_iteration(previous_walk_data=walk_data,
-                                           target_time=target_times[i])
+        walk_data = particle.run_iteration(target_time=target_times[i])
```

Comment:
When we re-initialize the `Particles` object, are we deleting the dictionary stored in `walk_data`, or is that still accessible? Wondering if we need to copy it

Reply:
So when we initialize a new `Particles` object, we lose all attributes of the old one. In this case, the old `walk_data` is fed into the generator in line 224 to pass the particle information to the various `Particles` that are initialized.

Reply:
Right, I'm just wondering if the `walk_data` on line 224 will still exist after we've deleted `self.walk_data` on line 219. Is the dict still stored in the linked external variable, or does it get wiped with the object?

Reply:
The `unsteady_example.py` script that uses this method works, so I guess the simple answer to this question is yes, the `walk_data` from line 224 continues to exist after line 219 is run.

I believe that the explanation for this is that line 219 does not actually "delete" `self.walk_data`. We are initializing a new `Particles` class, which I suspect gets its own, new position in memory, and so we don't lose the information in `walk_data` since we don't explicitly delete or remove it.